client/hotrod-client/src/main/java/org/infinispan/client/hotrod/CacheTopologyInfo.java

```diff
@@ -15,7 +15,7 @@
    /**
     * @return The number of configured segments for the cache.
     */
-   int getNumSegments();
+   Integer getNumSegments();
```

Comment:
`CacheTopologyInfo` is accessible from `RemoteCache.getCacheTopologyInfo()`, so this is going to break backwards compatibility.

Reply:
It was already broken, I didn't change the underlying code. Just the API definition. So the only thing I broke is binary compatibility as it already threw a NPE before :)

Reply:
https://github.com/infinispan/infinispan/blob/master/client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/CacheTopologyInfoImpl.java#L15